### PR TITLE
Get the real plural name from the CRD, rather than blindly appending 's'.

### PR DESCRIPTION
**Pull request by ggallen**
Resolves https://github.com/kubearchive/kubearchive/issues/136.

### CODE REVIEW

File: cmd/operator/internal/controller/kubearchiveconfig_controller.go [hunk 1]
```diff
@@ -162,18 +165,26 @@ func (r *KubeArchiveConfigReconciler) reconcileRole(ctx context.Context, kaconfi
...
-		slash := strings.Index(resource.APIVersion, "/")
-		if slash > -1 {
-			apiGroup = resource.APIVersion[:slash]
+		apiVersion := resource.APIVersion
+		data := strings.Split(resource.APIVersion, "/")
+		if len(data) > 1 {
+			apiGroup = data[0]
+			apiVersion = data[1]
+		}
+		gvr, err := r.Mapper.RESTMapping(schema.GroupKind{Group: apiGroup, Kind: resource.Kind}, apiVersion)
```
Comment (maruiz93) [inline cmd/operator/internal/controller/kubearchiveconfig_controller.go:179]: I think that we should add a comment to make it clear that gvr resources are expected to be the plural of the resource based on the kubernetes standard to use the plurals in the API endpoints.

Also, I think it's also worth to mention that k8s roles' rules expect the `resources` field to be filled with the plural lowercased version of the resource (as oposed to the singular PascalCase of the `kind` field of the resource)

**Review by maruiz93 — CHANGES_REQUESTED** [review]

File: cmd/operator/internal/controller/kubearchiveconfig_controller.go (hunk 1 above)
Comment (ggallen) [reply cmd/operator/internal/controller/kubearchiveconfig_controller.go:181]: Fixed.

**Review by skoved — COMMENTED** [review]
[The documentation for configuring KubeArchive](https://kubearchive.github.io/kubearchive/main/configuration/kubearchiveconfig.html) Shows a `KubeArchiveConfig` with the `kind` of the resource pluralized. There were a series of issues we were having around getting the `ApiServerSource` to created by the operator to work and in that process we switched to have the resource be singular. Do we want to stick with the singular (in which case the docs need to be updated) or should we switch back to specifying the plural to match the docs?

**Comment by ggallen** [comment]
@skoved, the `kind` should not be pluralized. It should be the real `kind` value.

I've updated the documentation. Nice catch.

**Review by maruiz93 — APPROVED** [review]

File: cmd/operator/internal/controller/kubearchiveconfig_controller.go [hunk 2]
```diff
@@ -162,18 +165,28 @@ func (r *KubeArchiveConfigReconciler) reconcileRole(ctx context.Context, kaconfi
...
+		// The resource field in the GVR contains the plural version of the resource, and
+		// the kubernetes Role expects this lower-cased plural version.
+		gvr, err := r.Mapper.RESTMapping(schema.GroupKind{Group: apiGroup, Kind: resource.Kind}, apiVersion)
+		if err == nil {
+			rules = append(rules, rbacv1.PolicyRule{
+				APIGroups: []string{apiGroup},
+				Resources: []string{strings.ToLower(gvr.Resource.Resource)},
+				Verbs:     []string{"get", "list", "watch"}})
+		} else {
+			log.Error(err, "Failed to get GVR for "+resource.APIVersion)
```
Comment (mafh314) [inline cmd/operator/internal/controller/kubearchiveconfig_controller.go:188]: WDYT about this: `log.Error("Failed to get GVR for: %s", resource.APIVersion, err)`

Comment (skoved) [reply cmd/operator/internal/controller/kubearchiveconfig_controller.go:188]: `[log.Error](https://pkg.go.dev/github.com/go-logr/logr#Logger.Error)` does not support format verbs, like `%s` and `logr.Logger` does not appear to have an `Errorf` method

Comment (ggallen) [reply cmd/operator/internal/controller/kubearchiveconfig_controller.go:188]: This PR has been merged already.

**Pull request merged**